homeassistant/components/lutron/fan.py

```diff
@@ -0,0 +1,123 @@
+"""Support for Lutron fans."""
+import logging
+
+from homeassistant.components.fan import (SPEED_LOW,
+                                          SPEED_MEDIUM,
+                                          SPEED_HIGH,
+                                          SPEED_OFF,
+                                          SUPPORT_SET_SPEED,
+                                          FanEntity)
+from . import LUTRON_CONTROLLER, LUTRON_DEVICES, LutronDevice
+
+_LOGGER = logging.getLogger(__name__)
+
+SPEED_MEDIUM_HIGH = 'medium_high'
+SPEED_MAPPING = {
+    SPEED_OFF: 0,
+    SPEED_LOW: 25,
+    SPEED_MEDIUM: 50,
+    SPEED_MEDIUM_HIGH: 75,
+    SPEED_HIGH: 100
+}
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Lutron fans."""
+    devs = []
+
+    for (area_name, device) in hass.data[LUTRON_DEVICES]['fan']:
+        dev = LutronFan(area_name, device, hass.data[LUTRON_CONTROLLER])
+        devs.append(dev)
+
+    add_entities(devs, True)
+
+
+class LutronFan(LutronDevice, FanEntity):
+    """Representation of a Lutron Fan, including dimmable."""
+
+    def __init__(self, area_name, lutron_device, controller):
+        """Initialize the fan."""
+        self._area_name = str(area_name)
+        self._name = str(area_name) + ' Fan'
+        self._is_on = False
+        self._speed = SPEED_OFF
+        self._prev_speed = None
+        super().__init__(area_name, lutron_device, controller)
```

Comment:
Move this first in the method body. The parent init method sets some attributes, eg `self._area_name`.